spec.bs

```diff
@@ -491,19 +491,61 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
-      1. Let |topics| be the result from running the [=calculate the topics for caller=] algorithm, with |topicsCallerContext| as input.
-      1. Let |headerStructuredFields| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-lists">List</a>.
-      1. Let |lastVersion| be null.
+      1. Let |topics| be the result of running the [=calculate the topics for caller=] algorithm, with |topicsCallerContext| as input.
+      1. Let |topicsInnerList| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-inner-lists">Inner List</a>.
+      1. Let |paramsSeparatorLength| be 2 (i.e. structured fields use two characters (';' and '=') for each parameter)
+      1. Let |paramKeyLength| be 1 (i.e. always use a single character parameter key)
+      1. Let |serializedTopicsInnerListLength| be 0.
       1. For each |topic| in |topics|:
+          1. Let |digitsInTopic| be the number of digits in |topic|["{{BrowsingTopic/topic}}"].
+          1. Increment |serializedTopicsInnerListLength| by |digitsInTopic|.
+          1. If |topicsInnerList| is not empty, then increment |serializedTopicsInnerListLength| by |paramsSeparatorLength|.
           1. Let |topicItem| be a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-integers">Integer</a> with value |topic|["{{BrowsingTopic/topic}}"].
           1. Let |topicParameters| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parameters">Parameters</a>.
-          1. If |lastVersion| is null, or if |lastVersion| does not equal |topic|["{{BrowsingTopic/version}}"], then set |topicParameters|["<code>v</code>"] to |topic|["{{BrowsingTopic/version}}"].
-          1. Set |lastVersion| to |topic|["{{BrowsingTopic/version}}"].
+          1. Increment |serializedTopicsInnerListLength| by the length of |topic|["{{BrowsingTopic/version}}"].
+          1. Increment |serializedTopicsInnerListLength| by |paramsSeparatorLength| + |paramKeyLength|.
+          1. Set |topicParameters|["<code>v</code>"] to a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-tokens">Token</a> having words |topic|["{{BrowsingTopic/version}}"].
           1. Associate |topicParameters| with |topicItem|.
-          1. Insert |topicItem| to |headerStructuredFields|.
+          1. Insert |topicItem| to |topicsInnerList|.
+      1. Let |maxNumberOfEpochs| be 3 (i.e. topics are selected from the last 3 epochs).
+      1. Let |topicMaxLength| be number of digits in the maximum [=browsing topics types/topic id=] (e.g. for <a href="https://github.com/patcg-individual-drafts/topics/blob/main/taxonomy_v1.md">Chrome's initial taxonomy</a>, |topicMaxLength| is 3, as the [=browsing topics types/topic id=] has maximum 3 digits).
+      1. Let |versionMaxLength| be the length of the current [=browsing topics types/version=].
```

Comment:
What is the "current" one? [=browsing topics types/version=] is only defined given the three inputs _configurationVersion_, _taxonomyVersion_, and _modelVersion_.

Probably the best solution here is to move this definition to https://patcg-individual-drafts.github.io/topics/#terminology-and-types-header to make it clear you're trying to figure out a user-agent-wide maximum. E.g. something like

> The maximum version string length is the maximum possible string length of a [=browsing topics types/version=] that a user agent could possibly generate in a given browsing session.

(or maybe in a given "software release"??)

and then adding an example would be even nicer.

Reply:
Done.